### DISCUSSION TITLE
Masonry with 3 columns and a sidebar?

### DISCUSSION THREAD

**Discussion by kreincke** [bootscore/bootscore#421] [Q&A]
Dear friends; could I get a little advice how I have to modify at archive-masonry.php if it shall deliver 3 columns and the sidebar constantly at the left size? Many thanks in advance KR

**Comment by crftwrk** [comment]
Do this:

```php
<?php

/**
 * Archive Template: Masonry
 *
 * @link https://developer.wordpress.org/themes/basics/template-hierarchy/
 *
 * @package Bootscore
 */

get_header();
?>
<div id="content" class="site-content container py-5 mt-5">
  <div id="primary" class="content-area">

    <!-- Hook to add something nice -->
    <?php bs_after_primary(); ?>

    <main id="main" class="site-main">

      <header class="page-header mb-4">
        <h1><?php the_archive_title(); ?></h1>
        <?php the_archive_description('<div class="archive-description">', '</div>'); ?>
      </header>
      
      <div class="row">
        
        <?php get_sidebar(); ?>
        
        <div class="col order-last">

          <div class="row" data-masonry='{"percentPosition": true }'>
            <?php if (have_posts()) : ?>
              <?php while (have_posts()) : the_post(); ?>

                <div class="col-md-6 col-lg-4 mb-4">

                  <div class="card">

                    <?php the_post_thumbnail('medium', array('class' => 'card-img-top')); ?>

                    <div class="card-body">

                      <?php bootscore_category_badge(); ?>

                      <h2 class="blog-post-title">
                        <a href="<?php the_permalink(); ?>">
                          <?php the_title(); ?>
                        </a>
                      </h2>

                      <?php if ('post' === get_post_type()) : ?>

                        <small class="text-muted mb-2">
                          <?php
                          bootscore_date();
                          bootscore_author();
                          bootscore_comments();
                          bootscore_edit();
                          ?>
                        </small>

                      <?php endif; ?>

                      <div class="card-text">
                        <?php the_excerpt(); ?>
                      </div>

                      <div class="">
                        <a class="read-more" href="<?php the_permalink(); ?>"><?php _e('Read more »', 'bootscore'); ?></a>
                      </div>

                      <?php bootscore_tags(); ?>

                    </div><!-- card-body -->

                  </div><!-- card -->

                </div><!-- col -->

              <?php endwhile; ?>
            <?php endif; ?>

          </div><!-- row -->
      
        </div>
      </div>

      <!-- Pagination -->
      <div>
        <?php bootscore_pagination(); ?>
      </div>

    </main><!-- #main -->

  </div><!-- #primary -->
</div><!-- #content -->
<?php
get_footer();

```

**Comment by kreincke** [comment]
cool. It works. It's a home-masonry-sidebar-left.php. I will now create a home-masonry-sidebar-right.php  - with only 3 columns. Thanks for your direct answer. KR

**Comment by kreincke** [comment]
Just for not turning your proposal into nonsense:

IF I 
* modify your line `<div class="col order-last">` into `<div class="col order-first">`
* replace your line `<div class="col-md-6 col-lg-4 col-xxl-3 mb-4">` by `<div class="col-md-6 col-lg-6 col-xl-4 mb-4">`
THEN I get, what I want to get:
* on screens of type < md (< sm) I get one column containing one post after another (plus closed sidebar and the opening button)
* on screens of type md <= x < lg I get two column containing posts plus an opened sidebar
* on screens of type lg <= x ... I get three columns containing posts plus an opened sidebar

If you want to see it please have a look at [https://karsten-reincke.de/](https://karsten-reincke.de/)

Did I understand your idea correctly? many thanks KR

**Comment by crftwrk** [comment]
> sidebar constantly at the left size

If you want to have sidebar on the left, than you have to keep class `order-last`, because sidebar has class `order-md-last` https://github.com/bootscore/bootscore/blob/main/sidebar.php

For masonry layout, it's mandatory to install the bS Masonry plugin https://github.com/bootscore/bs-masonry/archive/refs/heads/main.zip. If not installed, there are fixed rows instead masonry.

**Comment by kreincke** [comment]
Thanks for this answer! It was a 'typo'. I did not mean left, but right. So everything is ok. 

I will now modify 'our' archive-masonry-sidebar-right.php in the way, that a) it uses the_content(), instead of the_exzerpt() b) requires a more tag inserted on Gutenberg level c) does no longer use the 'more'-link set on template level. I think, that the masonry effect will become stronger, if the writer determines the length of the 'excerpt' - in accordance with the content.

If that works well, I will offer you both templates - that, as it exists now, and that, which I will create - as PRs for the template repository. 

With best regards KR
